streaming/src/main/scala/org/apache/spark/streaming/scheduler/ReceiverTracker.scala

```diff
@@ -137,8 +141,24 @@ class ReceiverTracker(ssc: StreamingContext, skipReceiverLaunch: Boolean = false
     }
     receiverInfo(streamId) = ReceiverInfo(
       streamId, s"${typ}-${streamId}", receiverActor, true, host)
+    logInfo(s"Receiver for stream ${streamId} from ${sender.path.address} " +
+      s"attempts to register.")
+  }
+
+  /** Report a starting receiver */
+  private def startingReceiver(
```

Comment:
@jerryshao IMHO, we only need the "ReceiverSupervisor" to register itself. Because, the restart() only restart the "process thread" inside the "ReceiverSupervisor", it dose not change any location information or state about the receiver. Only when a "receiver task", the restarted receiver needs to register itself again. So, we only need to register a receiver in the prestart of "ReceiverSupervisor". If I understand correctly.